pennylane/interfaces/batch/autograd.py

```diff
@@ -0,0 +1,170 @@
+
+
+def execute(tapes, device, execute_fn, gradient_fn, _n=1):
+    """Execute a batch of tapes with Autograd parameters on a device.
+
+    Args:
+        tapes (Sequence[.QuantumTape]): batch of tapes to execute
+        device (.Device): Device to use to execute the batch of tapes.
+            If the device does not provide a ``batch_execute`` method,
+            by default the tapes will be executed in serial.
+        execute_fn (callable): The execution function used to execute the tapes
+            during the forward pass. This function must return a tuple ``(results, jacobians)``.
+            If ``jacobians`` is an empty list, then ``gradient_fn`` is used to
+            compute the gradients during the backwards pass.
+        gradient_fn (callable): the gradient function to use to compute quantum gradients
+        _n (int): a positive integer used to track nesting of derivatives, for example
+            if the nth-order derivative is requested.
+
+    Returns:
+        list[list[float]]: A nested list of tape results. Each element in
+        the returned list corresponds in order to the provided tapes.
+    """
+
+    parameters = autograd.builtins.tuple(
+        [autograd.builtins.list(t.get_parameters()) for t in tapes]
+    )
+
+    return _execute(
+        parameters,
+        tapes=tapes,
+        device=device,
+        execute_fn=execute_fn,
+        gradient_fn=gradient_fn,
+        _n=_n,
+    )[0]
+
+
+@autograd.extend.primitive
+def _execute(
+    parameters, tapes=None, device=None, execute_fn=None, gradient_fn=None, _n=1
+):  # pylint: disable=dangerous-default-value,unused-argument
+    """Autodifferentiable wrapper around ``Device.batch_execute``.
+
+    The signature of this function is designed to workaround Autograd restrictions.
+    Note that the ``parameters`` argument is dependent on the ``tapes`` argument;
+    this function should always be called as follows:
+
+    >>> parameters = [autograd.builtins.list(t.get_parameters()) for t in tapes])
+    >>> parameters = autograd.builtins.tuple(parameters)
+    >>> _batch_execute(parameters, tapes=tapes, device=device)
+
+    In particular:
+
+    - ``parameters`` is dependent on the provided tapes: always extract them as above
+    - ``tapes`` is a *required* argument
+    - ``device`` is a *required* argument
+
+    The private argument ``_n`` is used to track nesting of derivatives, for example
+    if the nth-order derivative is requested. Do not set this argument unless you
+    understand the consequences!
+    """
+    with qml.tape.Unwrap(*tapes):
+        res, jacs = execute_fn(tapes)
+
+    return [np.tensor(r) for r in res], jacs
+
+
+def vjp(
+    ans, parameters, tapes=None, device=None, execute_fn=None, gradient_fn=None, _n=1
+):  # pylint: disable=dangerous-default-value,unused-argument
+    """Returns the vector-Jacobian product operator for a batch of quantum tapes.
+
+    Args:
+        ans (array): the result of the batch tape execution

```

Comment:
This is Autograd terminology from their docs. I'm somewhat tempted to keep it, so that 

- future developers can more easily cross-reference with the Autograd docs, and
- it retains the same terminology as the existing interfaces